<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TopologyRadiomicsSlicer</class>
 <widget class="qMRMLWidget" name="TopologyRadiomicsSlicer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>458</width>
    <height>909</height>
   </rect>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="parameterSetLabel">
     <property name="text">
      <string>Paremeter set: </string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="qMRMLNodeComboBox" name="parameterNodeSelector">
     <property name="toolTip">
      <string>Pick parameter set</string>
     </property>
     <property name="nodeTypes">
      <stringlist>
       <string>vtkMRMLScriptedModuleNode</string>
      </stringlist>
     </property>
     <property name="selectNodeUponCreation">
      <bool>true</bool>
     </property>
     <property name="addEnabled">
      <bool>true</bool>
     </property>
     <property name="renameEnabled">
      <bool>true</bool>
     </property>
     <property name="removeEnabled">
      <bool>true</bool>
     </property>
     <property name="noneEnabled">
      <bool>false</bool>
     </property>
     <property name="showHidden">
      <bool>true</bool>
     </property>
     <property name="showChildNodeTypes">
      <bool>false</bool>
     </property>
     <property name="baseName">
      <string>TopologyRadiomicsSlicer</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="ioCollapsibleButton">
     <property name="text">
      <string>Input</string>
     </property>
     <layout class="QFormLayout" name="basicCollapsibleFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Input Segmentation / Model:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="inputSelector">
        <property name="toolTip">
         <string>Pick the input to the algorithm - defined by a segmentation or model node.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLSegmentationNode</string>
         </stringlist>
        </property>
        <property name="noneEnabled">
         <bool>false</bool>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="showHidden">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="editEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="ctkCollapsibleButton" name="advancedCollapsibleButton">
     <property name="text">
      <string>Config</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <!-- Gaussian smoothing iterations -->
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_iter">
        <property name="text">
         <string>Gaussian smoothing iterations</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="ctkSliderWidget" name="iterSlider">
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>10</double>
        </property>
        <property name="pageStep">
         <double>1</double>
        </property>
        <property name="singleStep">
         <double>1</double>
        </property>
       </widget>
      </item>
      <!-- Gaussian smoothing sigma -->
      <item row="1" column="0">
       <widget class="QLabel" name="label_sigma">
        <property name="text">
         <string>Gaussian smoothing sigma</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkSliderWidget" name="sigmaSlider">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>5</double>
        </property>
        <property name="pageStep">
         <double>0.5</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <!-- Voxel spacing -->
      <item row="2" column="0" colspan="2">
       <widget class="QLabel" name="label_voxel_spacing">
        <property name="text">
         <string>Voxel spacing</string>
        </property>
       </widget>
      </item>
      <!-- Voxel spacing X -->
      <item row="3" column="0">
       <widget class="QLabel" name="label_voxel_spacing_x">
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="ctkSliderWidget" name="spacingXSlider">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>100</double>
        </property>
        <property name="pageStep">
         <double>1</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <!-- Voxel spacing Y -->
      <item row="4" column="0">
       <widget class="QLabel" name="label_voxel_spacing_y">
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="ctkSliderWidget" name="spacingYSlider">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>100</double>
        </property>
        <property name="pageStep">
         <double>1</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <!-- Voxel spacing Z -->
      <item row="5" column="0">
       <widget class="QLabel" name="label_voxel_spacing_z">
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="ctkSliderWidget" name="spacingZSlider">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>100</double>
        </property>
        <property name="pageStep">
         <double>1</double>
        </property>
        <property name="singleStep">
         <double>0.1</double>
        </property>
       </widget>
      </item>
      <!-- Marching Cubes Algorithm -->
      <item row="6" column="0">
       <widget class="QLabel" name="label_marching_cubes_algorithm">
        <property name="text">
         <string>Marching Cubes Algorithm</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="ctkComboBox" name="algorithmComboBox">
        <property name="editable">
         <bool>false</bool>
        </property>
        <property name="scrollWheelEffect">
         <string>NeverScroll</string>
        </property>
        <item>
         <property name="text">
          <string>Lewiner</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Lorensen</string>
         </property>
        </item>
       </widget>
      </item>
      <!-- Marching Cubes step size -->
      <item row="7" column="0">
       <widget class="QLabel" name="label_marching_cubes_step_size">
        <property name="text">
         <string>Marching Cubes step size</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="ctkSliderWidget" name="stepSizeSlider">
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>1</double>
        </property>
        <property name="maximum">
         <double>50</double>
        </property>
        <property name="pageStep">
         <double>5</double>
        </property>
        <property name="singleStep">
         <double>1</double>
        </property>
       </widget>
      </item>
      <!-- Clip percent -->
      <item row="8" column="0">
       <widget class="QLabel" name="label_clip">
        <property name="text">
         <string>Clip percent</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="ctkSliderWidget" name="clipSlider">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0</double>
        </property>
        <property name="maximum">
         <double>1</double>
        </property>
        <property name="pageStep">
         <double>0.1</double>
        </property>
        <property name="singleStep">
         <double>0.01</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QPushButton" name="applyButton">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="toolTip">
      <string>Compute topology</string>
     </property>
     <property name="text">
      <string>Run</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QTextBrowser" name="textBrowser">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;CoLlAGe captures subtle anisotropic differences in disease pathologies by measuring entropy of co-occurrences of voxel-level gradient orientations on imaging computed within a local neighborhood.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;CoLlAGe is based on the hypothesis that disruption in tissue microarchitecture can be quantified on imaging by measuring the disorder in voxel-wise gradient orientations. CoLlAGe involves assigning every image voxel a ‘disorder value’ associated with the co-occurrence matrix of gradient orientations computed around every voxel.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;After feature extraction, the subsequent distribution or different statistics such as mean, median, variance etc can be computed and used in conjunction with a machine learning classifier to distinguish similar appearing pathologies. The feasibility of CoLlAGe in distinguishing cancer from treatment confounders/benign conditions and characterizing molecular subtypes of cancers has been demonstrated in the context of multiple challenging clinical problems.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="acceptRichText">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QTextBrowser" name="textBrowser_2">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="html">
      <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Notes:&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The Haralick Window size default of -1 will result in a window size of SVD_RADIUS * 2 - 1.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The Difference Variance Interpretation means feature 10 has two interpretations, as the variance of |x-y| or as the variance of P(|x-y|).]&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Note that unlike 2D features, two dominant directions (θ 3D, φ 3D) are computed for 3D CoLlAGe associated with each c ∈ C, which provide complimentary information about the degree of disorder of the principal gradient orientations in (X,Y) and (X,Y,Z) directions.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="acceptRichText">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCheckBox</class>
   <extends>QCheckBox</extends>
   <header>ctkCheckBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkComboBox</class>
   <extends>QComboBox</extends>
   <header>ctkComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>TopologyRadiomicsSlicer</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TopologyRadiomicsSlicer</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>parameterNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>28</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>192</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
